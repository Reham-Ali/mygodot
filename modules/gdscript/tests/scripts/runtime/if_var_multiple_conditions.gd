func foo(n: int) -> bool:
	print("foo")
	return n > 0

func bar(n: int) -> int:
	print("bar")
	return n

func add(a: int, b: int) -> int:
	print("add")
	return a + b

func test_if(f: bool, a: int, b: int) -> void:
	print("--")
	if f, foo(a), var n := bar(b):
		print("t:%s" % n)
	elif var n = add(a, b), n >= 1:
		print("tt:%s" % n)
	else:
		print("f")

func test():
	if var x := 100:
		if x >= 1, var y := "ttt":
			print(y)
		print(100 + signi(x))

	test_if(false, 0, 0);
	test_if(false, 0, 1);
	test_if(false, 1, 0);
	test_if(false, 1, 1);
	test_if(true, 0, 0);
	test_if(true, 0, 1);
	test_if(true, 0, 2);
	test_if(true, 1, 0);
	test_if(true, 1, 1);
	test_if(true, 1, 2);
