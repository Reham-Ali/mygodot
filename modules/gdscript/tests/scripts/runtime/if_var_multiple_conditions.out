GDTEST_OK
ttt
101
--
add
f
--
add
tt:1
--
add
tt:1
--
add
tt:2
--
foo
add
f
--
foo
add
tt:1
--
foo
add
tt:2
--
foo
bar
add
tt:1
--
foo
bar
t:1
--
foo
bar
t:2
